extension/popup.js

```diff
@@ -1,147 +1,217 @@
-    }).catch(err => {
-      console.error("Failed to copy:", err);
-      alert("Failed to copy text to clipboard.");
-    });
-  });
-
-  // Get the video ID from the currently active YouTube tab
-  async function getCurrentVideoId() {
-    return new Promise((resolve) => {
-      chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
-        if (tabs.length > 0 && tabs[0].url) {
-          const url = new URL(tabs[0].url);
-          const videoId = url.searchParams.get("v");
-          console.log("Current video ID:", videoId);
-          resolve(videoId);
-        } else {
-          resolve(null);
+        try {
+            const response = await fetch(url);
+            const result = await response.json();
+            transcriptArea.value = result.status === "success" ? result.transcript : "Transcript not available.";
+        } catch (error) {
+            transcriptArea.value = `Error fetching transcript. ${error.message}`;
         }
-      });
-    });
-  }
+    },
+
+    async handleSummarize() {
+        await handlers.handleFetchTranscript();
+        const transcriptArea = document.getElementById("transcript-area");
+        const transcript = transcriptArea.value.trim();
+
+        if (!transcript) {
+            alert("Please fetch a transcript first!");
+            return;
+        }
+
+        const videoId = await utils.getCurrentVideoId();
+        const videoTitle = await utils.getVideoTitle(videoId);
+
+        await chrome.storage.local.set({
+            currentSummary: null,
+            summaryStatus: 'loading',
+            summaryError: null,
+            currentVideo: {
+                id: videoId,
+                title: videoTitle
+            },
+            initialTranscript: transcript
+        });
+
+        await utils.openPopupWindow('summary.html');
+
+        try {
+            const aiSettings = await api.getAiSettings();
+            const { systemPrompt } = await chrome.storage.sync.get(['systemPrompt']);
+
+            const response = await fetch(aiSettings.url, {
+                method: 'POST',
+                headers: { 'Content-Type': 'application/json' },
+                body: JSON.stringify({
+                    model: aiSettings.model,
+                    prompt: `System: ${systemPrompt || DEFAULT_SYSTEM_PROMPT}\n\nHuman: Here's a transcript, please summarize it:\n${transcript}\n\nAssistant:`,
+                    stream: false
+                })
+            });
+
+            const result = await response.json();
+            await chrome.storage.local.set({
+                currentSummary: result.response,
+                summaryStatus: 'complete'
+            });
+        } catch (error) {
+            await chrome.storage.local.set({
+                summaryStatus: 'error',
+                summaryError: `Failed to get summary. Please check the AI endpoint in settings. ${error.message}`
+            });
+        }
+    },
```

Comment:
Great suggestion, I've implemented the basic sanitation, however will wait until later when it comes to rate limiting. The idea of this extension is that it will run on a users local machine so the user will have control of the AI server as well and shouldn't really have to worry about rate limiting.